### PR TITLE
Cleanup before turnstile PR - thread getters

### PR DESCRIPTION
**Pull request by melouber**
Implemented getters.
I have no idea whether we want some other getters like `td_needs_switch` or `td_slice_ended`.
Maybe we want some setters like `td_set_ready`, `td_set_running`, `td_set_needswitch` or `td_clr_needswitch`. @cahirwpz ?

EDIT: 33th test seed fails. :confused: :confused:

### CODE REVIEW

**Comment by cahirwpz** [comment]
LGTM

**Comment by cahirwpz** [comment]
If @ilikeheaps accepts I'm willing to merge it.

File: include/mutex.h [hunk 1]
```diff
@@ -69,11 +69,6 @@ void mtx_unlock(mtx_t *m);
 
 DEFINE_CLEANUP_FUNCTION(mtx_t *, mtx_unlock);
 
-/*! \brief Mutex owner getter. */
-static inline volatile thread_t *mtx_owner(mtx_t *m) {
-  return m->m_owner;
-}
-
```
Comment (ilikeheaps) [inline include/mutex.h]: @cahirwpz weren't we supposed to replace `->m_owner` in tests with a getter?

Comment (melouber) [reply include/mutex.h]: (In our turnstile tests.)

Comment (cahirwpz) [reply include/mutex.h]: Ah... ok. I'm going to restore this function then.

**Pull request merged**